orca-spring-batch/src/main/java/com/netflix/spinnaker/zombie/ZombiePipelineCleanupAgent.java

```diff
@@ -0,0 +1,201 @@
+import com.netflix.spinnaker.orca.pipeline.model.Task;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.ApplicationContext;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import rx.Observable;
+import rx.Subscription;
+import static com.netflix.spinnaker.orca.ExecutionStatus.CANCELED;
+import static com.netflix.spinnaker.orca.ExecutionStatus.RUNNING;
+import static java.lang.String.format;
+import static java.time.Clock.systemDefaultZone;
+import static java.time.Duration.ZERO;
+import static java.time.Duration.between;
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE_TIME;
+import static rx.Observable.just;
+
+/**
+ * This poller cancels any pipelines that appear to be still running but are
+ * more than {@link #TOLERANCE} overdue.
+ */
+@Component
+@ConditionalOnExpression("${pollers.zombiePipeline.enabled:true}")
+@Slf4j
+public class ZombiePipelineCleanupAgent {
+
+  /**
+   * How often the agent runs.
+   */
+  private static final long FREQUENCY_MS = 21600000; // 6 hours
+
+  /**
+   * Delay before running the agent the first time.
+   */
+  private static final long INITIAL_DELAY_MS = 300000; // 5 minutes
+
+  private final ExecutionRepository repository;
+  private final Clock clock;
+  private final ApplicationContext applicationContext;
+  private final Lock lock;
+  private final String currentInstanceId;
+
+  /**
+   * Duration a task can be overdue before we decide to kill it.
+   */
+  static final Duration TOLERANCE = Duration.ofMinutes(30);
+
+  /**
+   * Duration a non-retryable task can run before being considered overdue.
+   * {@link #TOLERANCE} is added on.
+   */
+  static final Duration DEFAULT_TASK_TIMEOUT = Duration.ofMinutes(2);
+
+  @Autowired
+  public ZombiePipelineCleanupAgent(ExecutionRepository repository,
+                                    ApplicationContext applicationContext,
+                                    Lock lock,
+                                    String currentInstanceId) {
+    this(
+      repository,
+      applicationContext,
+      lock,
+      currentInstanceId,
+      systemDefaultZone()
+    );
+  }
+
+  ZombiePipelineCleanupAgent(ExecutionRepository repository,
+                             ApplicationContext applicationContext,
+                             Lock lock,
+                             String currentInstanceId,
+                             Clock clock) {
+    this.repository = repository;
+    this.clock = clock;
+    this.applicationContext = applicationContext;
+    this.currentInstanceId = currentInstanceId;
+    this.lock = lock;
+  }
+
+  @Scheduled(fixedDelay = FREQUENCY_MS, initialDelay = INITIAL_DELAY_MS)
```

Comment:
Don't we need something like `@EnableScheduling` on `ZombiePipelineCleanupConfig`, or is this scheduled some other way?